src/pathfinder/water_regions.cpp

```diff
@@ -49,7 +49,7 @@ class WaterRegionData {
 	std::array<TWaterRegionTraversabilityBits, DIAGDIR_END> edge_traversability_bits{};
 	std::unique_ptr<TWaterRegionPatchLabelArray> tile_patch_labels; // Tile patch labels, this may be nullptr in the following trivial cases: region is invalid, region is only land (0 patches), region is only water (1 patch)
 	bool has_cross_region_aqueducts = false;
-	TWaterRegionPatchLabel number_of_patches = 0; // 0 = no water, 1 = one single patch of water, etc...
+	TWaterRegionPatchLabel highest_assigned_label = INVALID_WATER_REGION_PATCH; // INVALID_WATER_REGION_PATCH = no water, FIRST_REGION_LABEL = one single patch of water, etc...
```

Comment:
Not an improvement. We want to know how many patches of water a region contains, that number is relevant for the neighbor search. The highest label is just an implementation detail, and even the fact that we label from low to high might change. Please rename it back to number_of_patches.

Also revert the comment change. It is much more clear to just have the numbers since you're talking about an amount, not a label.